### ISSUE TITLE
Lectern Causing Lag

### ISSUE THREAD

**Issue by Jonarlin** [Growlith1223/ArsMagica2#312]
The lectern causes lots of lag due to lots of chunk updates (sound this using sampler). When a lectern is placed it drops from 60 fps to 40 in the general lectern area.

**Comment by TheIcyOne** [comment]
Version?

**Comment by Jonarlin** [comment]
Sorry, I forgot. On 1.5.0-18.

**Comment by TheIcyOne** [comment]
Technically duplicate of #311, in that case, but it is probably cleaner to keep the issues separated.

**Comment by Haerar** [comment]
Possible link - FPS also drops when I place the Inscription Table; drop was from 80 fps to 16fps in version 1.10.2 1.5.0-18. Chunk updates went from 0-1 (prior to placing) to an instant 35-36 when placing the table. Did not move during this time; so not caused by world loading.

Both of the above perfectly reproducible on my end; remove table, issue disappears, place table, issue reappears - doesn't matter which chunk it's placed in. With my (admittedly limited) modding knowledge, I suspect that a loop that both of these tileentities run through needs to have a modulo applied; to reduce chunk update rate.

**Comment by TheIcyOne** [comment]
I thought I'd already fixed that o.O

**Comment by Growlith1223** [comment]
i may have left a chunk call in the code in that block. regardless, im unable to fix it for a few more days, until friday it won't be able to be fixed by me. the laptop im on is low-end so i doubt i could run ars magica on here lol. but once it's friday, it should be fixed that day or the next(depending on when the cable company arrives to add internet to the house.)

**Comment by Haerar** [comment]
Right, bit more information after some experimentation... the issue only arises when placed next to a bibliocraft table; so I'll remove those and we can consider my contributions null and void x.x oddly, bibliocraft's table and the inscription table only seem to dislike each other... placing anything else next to the other doesn't cause a problem. False alarm all! sorry!

Lectern still causing it's issues but I'll experiment with different materials/blocks near it and see if this has any bearing.

@TheIcyOne - can I ask what caused the inscription table issue originally? I am well-versed in java and have dabbled with minecraft modding; made a fun zombie mod once... wonder if I can spot anything perhaps?

EDIT: Any tileentity from AM2 and a bibliocraft table will cause issues next to each other; not sure why but still experimenting. Other transparent blocks from bibliocraft have no issue with AM2 tileentities, bibliocraft tables have no issue with tileentities from other mods (enderio, for example) , and any regular AM2 blocks (witchwood planks, for example) have no impact either way.

**Comment by Haerar** [comment]
Right, bit more information after some experimentation... the issue only arises when placed next to a bibliocraft table; so I'll remove those and we can consider my contributions null and void x.x oddly, bibliocraft's table and the inscription table only seem to dislike each other... placing anything else next to the other doesn't cause a problem. False alarm all! sorry!

Lectern still causing it's issues but I'll experiment with different materials/blocks near it and see if this has any bearing.

@TheIcyOne - can I ask what caused the inscription table issue originally? I am well-versed in java and have dabbled with minecraft modding; made a fun zombie mod once... wonder if I can spot anything perhaps?

**Comment by TheIcyOne** [comment]
Original issue was just that a few update functions were being called unnecessarily several times a tick.

**Comment by cool2412** [comment]
Is there a fix for this out? because my friends and I are running a server with Ars Magica 2 version 1.10.2-1.5.0-18 and My friend gets 108 FPS with the lectern removed but only 18 FPS when it is placed down.

**Comment by TheIcyOne** [comment]
Downgrade to the previous version, the issue should be new.

**Comment by scottosx** [comment]
I have verified that both the Lecturn and the Inscription table cause excessive chunk updates when placed. Both in 1.5.17b and 1.5.18. There are between 50 and 100 constant chunk updates. This is in a superflat world (quartz slab surface) with no other structures or blocks. I lose about 10fps.

Looking away from either of these items normalizes chunk updating. Anytime either of these items are in your render distance the problem occurs.

It sounds like a fix is known. Can this be prioritized? It is a significant perfomance issue that needs addressed as these are basic items needed to play the mod.

**Comment by scottosx** [comment]
Ok, so I was working with Simply Magic 2 v0.14 as my pack, which includes Ars Magica 2 1.5.17b. I am remixing it to my satisfaction. So, all my worlds were based on it. Just now I exported the profile from the Twitch launcher and reinstalled all mods leaving only the configs and scripts.

I don't have the frame drops anymore in new worlds created with the fresh pack! All chunk updates are normal now. Could it be that a previous version of AM2 still had data stored that creating the issue? In the new pack I updated to 1.5.18. I did in the old pack too but it didn't fix anything.

This might no longer be an issue with clean installs. What data could be responsible is the question.

**Comment by TheIcyOne** [comment]
I have a feeling the fix has been committed, just waiting on a release.

**Comment by D0gee** [comment]
Can this be prioritized please? Having lectern placed drops my FPS from 100 to 10.

**Comment by StalkerM8** [comment]
So anyone actually fix this? Its still relevant in 1.10.2. 
On a server running same versions
